InterfaceFortran/tsgC2Fortran.cpp

```diff
@@ -104,30 +116,73 @@ void tsgwri_(int *id, const char *filename, int *binary){ _tsg_grid_list[*id]->w
 void tsgrea_(int *id, const char *filename){ _tsg_grid_list[*id]->read(filename); }
 
 // create
-void tsgmg_(int *id, int *dimensions, int *outputs, int *depth, int *type, int *rule, const int *anisotropic_weights, double *alpha, double *beta, const int *llimits){
+void tsgmg_(int *id, int *dimensions, int *outputs, int *depth, int *type, int *rule, int *opt_flags,
+            const int *aniso_weights, double *alpha, double *beta, const char *customRuleFilename, const int *llimits){
+    double al, be;
+    const int  *aw, *ll;
+    const char *cfn;
+
+    opt_flags[0] == 1 ? aw  = aniso_weights : aw = 0;
+    opt_flags[1] == 1 ? al  = *alpha : al = 0;
```

Comment:
why not:
```
al = (opt_falgs[1] != 0) *alpha : 0.0;
```
* this makes it explicit that you are assigning something to `al`
* also, to avoid confusion, I always use `0.0` for floats

Reply:
You are right, this is better style.